packages/PowerGrid/src/fit/fit-curve.ts

```diff
@@ -0,0 +1,240 @@
+    if (!coordinateDf) return;
+
+    const coordinateColumns = coordinateDf.columns.byNames(['x', 'y']);
+    const coordinates: {x: number[], y: number[]} = {x: coordinateColumns[0].toList(),
+      y: coordinateColumns[1].toList()};
+
+    // this to call rerendering??
+    const updatedWidth = 160;
+    gridCell.gridColumn.width = updatedWidth;
+
+    const canvasCoordinates = scaleCoordinates(coordinates, gridCell.bounds.width, gridCell.bounds.height);
+
+    const markerPxSize = 6;
+    for (let i = 0; i < canvasCoordinates.x.length; i++) {
+      const markerCenterX = gridCell.bounds.x + canvasCoordinates.x[i];
+      const markerCenterY = (gridCell.bounds.y + gridCell.bounds.height) - canvasCoordinates.y[i];
+      if (e.offsetX > markerCenterX - (markerPxSize / 2) && e.offsetX < markerCenterX + (markerPxSize / 2) &&
+          e.offsetY > markerCenterY - (markerPxSize / 2) && e.offsetY < markerCenterY + (markerPxSize / 2))
+        coordinateDf.filter.set(i, !(coordinateDf.filter.get(i)));
+    }
+
+    grok.shell.o = gridCell;
+  }
+
+  render(
+    g: CanvasRenderingContext2D,
+    x: number, y: number, w: number, h: number,
+    gridCell: DG.GridCell, cellStyle: DG.GridCellStyle
+  ) {
+    // randomize dataframe columns with x and y (grok.data.demo.fit()?)
+    // set x like this higher (fixed)
+    // set y randomized between [-1; 1]
+    // make all markers the same color (like in scatterplot)
+    // fix max-min range, make width for renderer cell ??
+    // set bigger size for renderer (width height of cell ~ 200x100 and a little demo) !!!!
+    // make outliers switch (below)!
+    // TODO: add filtering bitset for points (if clicked for example)
+    // on click turn off the point (change bitset on false) - change it to cross (DG.MARKER_TYPE??)
+    // second click returns it
+    // fix demo dataset (parameteres?)
+    // make viewer (on selected rows with the name??)
+
+
+    // make normal resizing (width height - make with bounds) ???
+    // that's easy - just redefine the defaultHeight and etc. (we can do this)
+
+    // add confidence intervals (top and bottom)
+    // ask Leonid about why confidenceTop, confidenceBottom and fittedCurve give the same value
+
+    // make viewer using echarts?? (interactive, synchronizing with the renderer)
+    // make markers and lines one color on viewer (blue - blue, green - green, etc.)
+
+    // curves have to be like dose response curve
+
+    const data = getChartData(gridCell);
+
+    const df = gridCell.grid.dataFrame;
+    if (w < 20 || h < 10 || df === void 0) return;
+
+    const dfColumn = df.columns.toList().filter((column) => column.type === DG.COLUMN_TYPE.DATA_FRAME)![0];
+    const coordinateDf = DG.toJs(dfColumn.get(gridCell.cell.row.idx)) as DG.DataFrame;
+    if (!coordinateDf) return;
+
+    const coordinateColumns = coordinateDf.columns.byNames(['x', 'y']);
+    const coordinates: {x: number[], y: number[]} = {x: coordinateColumns[0].toList(),
```

Comment:
Getting coordinates seem to be a common operation for this renderer. Consider creating a separate function for that.